### PR TITLE
Add tests for two argument open and update docs

### PR DESCRIPTION
**Pull request by Quipyowert2**
# Description

Please include a summary of the proposed improvement or addressed issue.

Added tests for two-argument `open || die`, and updated documentation.

Fixes/addresses (If applicable) #3 (issue)

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project, please see the [contribution guidelines](https://github.com/jonasbn/perl-date-holidays/blob/master/CONTRIBUTING.md).
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
  The tests I added cause "Literal line breaks in a string" warnings with `perlcritic --top t/test.t`
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

You might think, that this is one crazy checklist, but it is just as much written for the maintainer of the involved software :-)

### CODE REVIEW

**Comment by jonasbn** [comment]
Closes #3

File: t/test.t [hunk 1]
```diff
@@ -39,7 +39,7 @@ my $critic = Perl::Critic->new(
 
     my @violations = $critic->critique( \$str );
 
-    is( scalar @violations, 0 );
+    is( scalar @violations, 0, "No bug - with parenthesis");
```
Comment (jonasbn) [inline t/test.t:42]: Please use single-quotes for non-interpolated strings as described in [ProhibitInterpolationOfLiterals](https://metacpan.org/pod/Perl::Critic::Policy::ValuesAndExpressions::ProhibitInterpolationOfLiterals)

```suggestion
    is( scalar @violations, 0, 'No bug - with parenthesis' );
```

File: t/test.t [hunk 2]
```diff
@@ -63,7 +63,7 @@ my $critic = Perl::Critic->new(
 
     my @violations = $critic->critique( \$str );
 
-    is( scalar @violations, 0 );
+    is( scalar @violations, 0, "No bug - with or and parenthesis");
```
Comment (jonasbn) [inline t/test.t:66]: Please use single-quotes for non-interpolated strings as described in [ProhibitInterpolationOfLiterals](https://metacpan.org/pod/Perl::Critic::Policy::ValuesAndExpressions::ProhibitInterpolationOfLiterals)

```suggestion
    is( scalar @violations, 0, 'No bug - with or and parenthesis' );
```

File: t/test.t [hunk 3]
```diff
@@ -75,7 +75,67 @@ my $critic = Perl::Critic->new(
...
+# The bug with two-arg open
+{
+    my $str = q[
+        open my $fh, "<$file"
+            || die "Can't open '$file': $!";
+    ];
+
+    my @violations = $critic->critique( \$str );
+
+    is( scalar @violations, 1, "Two-arg open bug");
```
Comment (jonasbn) [inline t/test.t:90]: Please use single-quotes for non-interpolated strings as described in [ProhibitInterpolationOfLiterals](https://metacpan.org/pod/Perl::Critic::Policy::ValuesAndExpressions::ProhibitInterpolationOfLiterals)

```suggestion
    is( scalar @violations, 1, 'Two-arg open bug' );
```

File: t/test.t [hunk 4]
```diff
@@ -75,7 +75,67 @@ my $critic = Perl::Critic->new(
...
+# No bug - with parenthesis
+{
+    my $str = q[
+        open(my $fh, "<$file")
+            || die "Can't open '$file': $!";
+    ];
+
+    my @violations = $critic->critique( \$str );
+
+    is( scalar @violations, 0, "No bug - with parenthesis");
```
Comment (jonasbn) [inline t/test.t:102]: Please use single-quotes for non-interpolated strings as described in [ProhibitInterpolationOfLiterals](https://metacpan.org/pod/Perl::Critic::Policy::ValuesAndExpressions::ProhibitInterpolationOfLiterals)

```suggestion
    is( scalar @violations, 0, 'No bug - with parenthesis' );
```

File: t/test.t [hunk 5]
```diff
@@ -75,7 +75,67 @@ my $critic = Perl::Critic->new(
...
+# No bug - with or
+{
+    my $str = q[
+        open my $fh, "<$file"
+            or die "Can't open '$file': $!";
+    ];
+
+    my @violations = $critic->critique( \$str );
+
+    is( scalar @violations, 0, "No bug - with or");
```
Comment (jonasbn) [inline t/test.t:114]: Please use single-quotes for non-interpolated strings as described in [ProhibitInterpolationOfLiterals](https://metacpan.org/pod/Perl::Critic::Policy::ValuesAndExpressions::ProhibitInterpolationOfLiterals)

```suggestion
    is( scalar @violations, 0, 'No bug - with or' );
```

File: t/test.t [hunk 6]
```diff
@@ -75,7 +75,67 @@ my $critic = Perl::Critic->new(
...
+# No bug - with or and parenthesis
+{
+    my $str = q[
+        open(my $fh, "<$file")
+            or die "Can't open '$file': $!";
+    ];
+
+    my @violations = $critic->critique( \$str );
+
+    is( scalar @violations, 0, "No bug - with or and parenthesis");
```
Comment (jonasbn) [inline t/test.t:126]: Please use single-quotes for non-interpolated strings as described in [ProhibitInterpolationOfLiterals](https://metacpan.org/pod/Perl::Critic::Policy::ValuesAndExpressions::ProhibitInterpolationOfLiterals)

```suggestion
    is( scalar @violations, 0, 'No bug - with or and parenthesis' );
```

File: t/test.t [hunk 7]
```diff
@@ -75,7 +75,67 @@ my $critic = Perl::Critic->new(
...
+# No bug - with or and parenthesis and whitespace
+{
+    my $str = q[
+        open ( my $fh, "<$file" )
+            || die "Can't open '$file': $!";
+    ];
+
+    my @violations = $critic->critique( \$str );
+
+    is( scalar @violations, 0, "No bug - with or and parenthesis and whitespace");
```
Comment (jonasbn) [inline t/test.t:138]: Please use single-quotes for non-interpolated strings as described in [ProhibitInterpolationOfLiterals](https://metacpan.org/pod/Perl::Critic::Policy::ValuesAndExpressions::ProhibitInterpolationOfLiterals)

```suggestion
    is( scalar @violations, 0, 'No bug - with or and parenthesis and whitespace' );
```

**Review by jonasbn — CHANGES_REQUESTED** [review]
Thanks for the contribution, I have some cosmetic issues, I want you to address

**Comment by jonasbn** [comment]
I am sorry about the long wait, but since not release has never been made this was not really on my radar. I had created a PR for inclusion into Perl::Critic, this PR has however open for  a long time, so I have retracted it and plan to release this as a stand-alone policy, hoping to get your contribution included.

**Comment by Quipyowert2** [comment]
I have changed the test names to use single quotes as requested.

**Review by jonasbn — APPROVED** [review]

**Comment by jonasbn** [comment]
Hi @Quipyowert2 

Thank you for your contribution, I will prepare and upload a release to PAUSE/CPAN shortly.

Take care

**Pull request merged**